### PR TITLE
Another try at dockerversion placeholder for library import

### PR DESCRIPTION
**Pull request by vdemeester**
> Hope it does not broke everything again :scream: 

On this one decided to move to the `dockerversion` package <del>— looked accross other `docker/*` project and almost all of them are using this package for versions (`swarm`, `distribution`, `libcompose`, …) — and decided to rename `VERSION` to `VERSION.txt`</del> 🐍.
- Ping @cpuguy83 @icecrime @jhowardmsft and any other folks that is building docker something else than Linux 🐧 for testing this on your computers (until I get my hands on some windows or apple computers xD).
  <del>\- Ping @tianon @jfrazelle on the `VERSION` to `VERSION.txt` files. I've updated docs and `rules` (for debian) that refers to this file but it might be used elsewhere — and if we can't rename that one, we'll find another package :wink: .</del>

<del>I'll squash the commits if the move `VERSION` to `VERSION.txt` is ok.</del>

To re-add some context :

> what about doing this, instead, as a placeholder? https://gist.github.com/erikh/e6f0c5e4b7b6c2483d52
> 
> you could leave this in the repository so it's still go-gettable, but then overwrite it during the build. how does this sound?

To add to the discussion, quotation from @ibuildthecloud 

> I know people shouldn't import non-pkg code but I don't like that effectively docker makes it not possible. If people want to import stuff at their own risk, just let them.

🐸

### CODE REVIEW

**Comment by tianon** [comment]
The "VERSION" filename is a pretty well-established convention across
projects, so doing this just because we can't think of a better
name/location for the version package feels wrong IMO.  We've also got a
lot of people using this file in scripting outside our repo (both packaging
and otherwise) so this might cause some backlash there.

The "version" package was "dockerversion" before it moved into autogen fwiw.

**Comment by vdemeester** [comment]
@tianon feels ok to me to keep `dockerversion` name, or even `autogen/dockerversion` :stuck_out_tongue_closed_eyes: (it feels safer and less changes too :wink: ).

**Comment by vdemeester** [comment]
ping @cpuguy83 @icecrime @jhowardmsft for trying this out on case-insensitive fs :cat: 
ping @jfrazelle @tianon @calavera  for LGTMs :kissing_cat:

**Comment by jhowardmsft** [comment]
Yay, it compiles. I think you need to add dockerversion/ to .gitignore though. 
it didn t break

**Comment by vdemeester** [comment]
Ohh right, true :wink:

**Comment by jhowardmsft** [comment]
LGTM otherwise.

File: hack/make/.go-autogen [hunk 1]
```diff
@@ -2,20 +2,23 @@
...
+// This file is overridden on build with build-time informations.
+const (
+	GitCommit string = "$GITCOMMIT"
+	Version   string = "$VERSION"
+	BuildTime string = "$BUILDTIME"
 
-	IAMSTATIC string = "${IAMSTATIC:-true}"
-	INITSHA1  string = "$DOCKER_INITSHA1"
-	INITPATH  string = "$DOCKER_INITPATH"
+	IAmStatic string = "${IAMSTATIC:-true}"
```
Comment (calavera) [inline hack/make/.go-autogen]: should this check the env variable in camel case beside in all capitals? I guess we left it like that to be backwards compatible, but should we also be consistent?

Comment (tianon) [inline hack/make/.go-autogen]: Environment variables are traditionally all-caps, so I'm pretty ¯\_(ツ)_/¯ on this one. :+1:

**Comment by tianon** [comment]
LGTM pending `.gitignore` entry updating :+1:

File: hack/make/.go-autogen (hunk 1 above)
Comment (vdemeester) [inline hack/make/.go-autogen]: Just like @tianon, didn't want to mess up with env' variables so ¯_(ツ)_/¯

Comment (calavera) [inline hack/make/.go-autogen]: sounds good, I only wanted to double check :smile_cat:

**Comment by calavera** [comment]
LGTM

**Pull request merged**